test/libsolidity/syntaxTests/array/length/parameter_too_large_multidim.sol

```diff
@@ -2,10 +2,8 @@ contract C {
   function f(bytes32[1263941234127518272][500] memory) public pure {}
   function f(uint[2**30][] memory) public pure {}
   function f(uint[2**30][2**30][] memory) public pure {}
-  function f(uint[2**16][2**16][] memory) public pure {}
```

Comment:
What about this one?

Reply:
I'm not sure actually... the bounds are/were quite arbitrary, weren't they? The change from `validForCalldata` to `validForMemory` affected this, I think - I don't think it's really a problem...

Reply:
Can you decrease the bound and see when it works?

Reply:
Ah sorry, this is a different one: Why did you remove it? :)

Reply:
It's removed, since it wouldn't fail with the new logic :-).

Reply:
ok!

Reply:
Hm... maybe there is actually a problem here...

Reply:
Yeah, this needs some fixing - I'll look into it.

Reply:
The idea here was also that big memory objects should be disallowed in the first place, so even though the individual objects / dimensions in e.g. `uint[10][10][10][10][10][10][10][10][10][10][10][10]` are rather small, creating a single object of this type should not possible and detected at compile-time.

Reply:
I strengthened the memory array checks a bit further to sum up nested statically sized parts.